crates/re_query/src/range/query.rs

```diff
@@ -47,7 +48,65 @@ impl Caches {
             };
 
             let mut cache = cache.write();
+
+            // TODO(#4810): Get rid of this once we have proper bucketing in place.
+            //
+            // Detects the case where the user loads a piece of data at the end of the time range, then a piece
+            // at the beginning of the range, and finally a piece right in the middle.
+            //
+            // DATA = ###################################################
+            //          |      |     |       |            \_____/
+            //          \______/     |       |            query #1
+            //          query #2     \_______/
+            //                       query #3
+            //
+            // and coarsly invalidates the whole range in that case, to avoid the kind of bugs
+            // showcased in <https://github.com/rerun-io/rerun/issues/5686>.
+            {
+                let time_range = cache.per_data_time.read_recursive().time_range();
+                if let Some(time_range) = time_range {
+                    {
+                        let hole_start = time_range.max();
+                        let hole_end =
+                            TimeInt::new_temporal(query.range().min().as_i64().saturating_sub(1));
+                        if hole_start < hole_end {
+                            if let Some((data_time, _, _)) = store.latest_at(
+                                &LatestAtQuery::new(query.timeline(), hole_end),
+                                entity_path,
+                                component_name,
+                                &[component_name],
+                            ) {
+                                if data_time != hole_start {
```

Comment:
Again a little confused by what's happening here. I suspect this is to avoid invalidating in cases where the hole otherwise contains no data.

However, shouldn't it be `data_time > hole_start`. 

Any situation where `data_time < hole_start` doesn't seem like it should require an invalidation since the values pulled in by the new query should be correct?

Reply:
Yeah you're right, nice catch